canonical/loki-k8s-operator
# Integrate Loki with Traefik ingress

## Issue

This PR fixes #116 and #117 (since they are related)


## Solution


This PR integrates `ingress_per_unit` lib with `Loki`

## Context


Loki send over relation data its endpoint (internal cluster IP) so client charms can set up it, but this is not useful for clients that are outside the cluster. That why we are integrating `ingress_per_unit` lib from Traefik charm

## Testing Instructions

1. Deploy Grafana agent charm
2. Deploy Loki charm
3. Deploy Traefik charm
4. Add a relation between Grafana agent and Loki
5. Check in `/etc/agen/agent.yaml` that Loki's endpoint is present.
6. Relate Loki with Traefik
7. Check in `/etc/agen/agent.yaml` that Traefik based endpoint is present.

## Release Notes
